### DISCUSSION TITLE
我有三个新的想法

### DISCUSSION THREAD

**Discussion by s192516** [Flow-Launcher/Flow.Launcher#1170] [Ideas]
Hello, I have three new ideas
1. Dynamically adjust the command sequence
Search a file, right click into the context menu, the order of additional information is always fixed, whether it can be adjusted to dynamic, such as allowing users to manually adjust or adjust according to the frequency of use, I often use the function of copying files, every time I have to look down. The atL +7 shortcut is also less useful than ATL +1
2. Quick search
We often give nicknames or abbreviations to commonly used documents or programs, such as "household accounting record.doc" for short "order" or "hf". This is just an example of how different languages and people have different abbreviations, and these abbreviations or nicknames are not searchable by autocomplete. I want to add a new feature that allows users to search for a particular file by customizing keywords
3. Add new commands. Sometimes we look for a file not only to open it, but to compress or print it, as well as other functions. It is very complicated to implement, so I recommend keeping things simple, leaving out some parameters that require user input, giving only the default common options, and letting the user use the original method if he needs something more complicated

thank you

--------

您好, 我有三个新的想法 
1. 动态调整指令顺序 
搜索到某个文件, 右键进入上下文菜单后,  附加信息顺序永远是固定的, 是否可以调整为动态的, 例如允许用户手动调整或者根据使用频率调整, 我自己非常常用复制文件功能, 每次都要向下找. 快捷键atl+7也没有 atl+1好用 
2. 快捷搜索 
我们经常会给一些常用的文件或程序起一个昵称或简称, 例如"家庭财务收支记录表.doc"简称为"order" 或者"hf", 这只是一个举例不同语言不同人有不同的缩略习惯, 这些简写或昵称是无法通过自动补全搜索到的, 希望可以添加一个新的功能, 让用户可以通过自定义关键词来搜索到某个特定文件 
3.添加新的指令, 有时候我们查找某个文件不仅是为了打开,而是压缩或者打印,   以及一些其他的功能, 希望可以加入类似于linux中" zip test.dox" 类似的指令, 这个功能可能会有很多细节需要考虑的地方, 实现起来非常复杂, 我建议一切从简, 省略一些需要用户输入的参数, 只给默认常用选项, 如果用户需要更复杂的功能让他使用原来的方法来实现

**Comment by taooceros** [comment]
Move to Discussion.
1.  - I agree with you, but somehow it is hard to do it. Currently all the context menu is generated dynamically from plugins, and so it will be very hard to implement a manually adjustable context menu. 
    - Dynamic sorting based on selected behavior is possible, since we already implement that for the main result panel, and I think it is reasonable to have that for context menu.

2. In progress https://github.com/Flow-Launcher/Flow.Launcher/pull/930
3. You may want this feature? We can make that more accessible in the future. ![image](https://user-images.githubusercontent.com/45326534/166239829-a4195e11-deb6-433f-aa41-810476c1267b.png)

**Reply by jjw24** [reply]
> 3. You may want this feature? We can make that more accessible in the future.

My concern with this whilst it's easy to implement, it does look out of place and a bit 'hacky' with the overall aesthetic of flow. What about listing the top needed options and add them as we go?
